utils/ibdmp-decode

```diff
@@ -0,0 +1,261 @@
+        "Decode debug tarball emited by leapp's initramfs in-band",
+        "console debugger, ibdmp().",
+    ]
+    sys.stderr.writelines('%s\n' % l for l in lines)
+    sys.exit(2)
+
+
+def LOG_DEBUG(msg):
+    if DEBUG:
+        sys.stderr.write('DEBUG: %s\n' % msg)
+
+
+def LOG_WARN(msg):
+    sys.stderr.write('%s\n' % msg)
+
+
+class IbdmbDecodeError(ValueError):
+    pass
+
+
+class UsageError(ValueError):
+    pass
+
+
+class _Chunk:
+
+    @classmethod
+    def from_raw1(cls, raw_chunk):
+        assert raw_chunk.startswith('_ibdmp:')
+        areas = raw_chunk.split('|')
+        parts = areas[1].split(':')
+        return cls(
+            ordinal=int(parts[0]),
+            payload=str(parts[1]),
+        )
+
+    def __init__(self, ordinal, payload):
+        self.ordinal = ordinal
+        self.payload = payload
+
+    def __hash__(self):
+        return hash((self.ordinal, self.payload))
+
+
+class Header:
+    """
+    Chunk set header
+    """
+
+    @classmethod
+    def from_rawN(cls, raw_headers):
+        """
+        Initialize chunk header from header chunk candidates
+
+        raw_headers is a list of strings that contain encoded chunk
+        parameters for the whole chunk set, ie. number of chunks, number
+        of iterations, and MD5 hash of the content encoded in the chunk set.
+
+        Raw header chunks can be corrupted so this factory will choose
+        winner based on prevalence.
+
+        For chunk set example in ChunkCounter.__init__ corresponding
+        raw headers could look similar to this:
+
+            _ibdmp:1/3|chunks=2,md5=281cc34e13cb4a502abd340fd07c4020|
+            _ibdmp:2/3|chunks=2,md5=281cc34e13cb4a502abd340fd07c4020|
+            _ibdmp:3/3|chun?s=2,md5=281cc34e13cb4a502abd340fd07c4020|
+
+        In this case, the winner is the first and second one.
+        """
+        cntr = collections.Counter([
+            Header._from_raw1(rh) for rh in raw_headers
+        ])
+        if not cntr:
+            LOG_WARN("no dumps found in this console log")
+            raise IbdmbDecodeError()
+        winner = cntr.most_common()[0][0]
+        LOG_DEBUG("header winner: %s" % winner)
+        return winner
+
+    @classmethod
+    def _from_raw1(cls, raw_header):
+        parts = raw_header.split('|')
+        _, stats = parts[0].split(':')
+        pairs = parts[1].split(',')
+        try:
+            assert pairs[0].startswith('chunks=')
+            assert pairs[1].startswith('md5=')
+        except AssertionError:
+            LOG_WARN("invalid header chunk payload: %s"
+                     % raw_header)
+            raise IbdmbDecodeError(raw_header)
+        return cls(
+            chunks=int(pairs[0].split('=')[1]),
+            md5=str(pairs[1].split('=')[1]),
+            csets=int(stats.split('/')[1]),
+        )
+
+    def __init__(self, chunks, md5, csets):
+        self.chunks = chunks
+        self.md5 = md5
+        self.csets = csets
+
+    def __eq__(self, othr):
```

Comment:
Make sure to also provide `__neq__` to save some headaches later on (right, @pirat89 ? :))

Reply:
I was under the impression that one does not need to do that in Python 3 anymore... but I might be wrong.  I've added `__neq__` in abf4c26